magenta/music/musicxml_parser.py

```diff
@@ -0,0 +1,678 @@
+  def __init__(self, xml_note, state):
+    self.xml_note = xml_note
+    self.voice = 1
+    self.is_rest = False
+    self.is_in_chord = False
+    self.is_grace_note = False
+    self.pitch = None               # Tuple (Pitch Name, MIDI number)
+    self.note_duration = NoteDuration(state)
+    self.state = state
+    self.parse()
+
+  def parse(self):
+    """Parse the MusicXML <note> element"""
+
+    self.midi_channel = self.state.midi_channel
+    self.midi_program = self.state.midi_program
+    self.velocity = self.state.velocity
+
+    for child in self.xml_note:
+      if child.tag == "chord":
+        self.is_in_chord = True
+      elif child.tag == "duration":
+        self.note_duration.parseduration(self, child.text)
+      elif child.tag == "pitch":
+        self.parsepitch(child)
+      elif child.tag == "rest":
+        self.is_rest = True
+      elif child.tag == "voice":
+        self.voice = int(child.text)
+      elif child.tag == "dot":
+        self.note_duration.dots += 1
+      elif child.tag == "type":
+        self.note_duration.type = child.text
+      elif child.tag == "time-modification":
+        # A time-modification element represents a tuplet_ratio
+        self.parsetuplet(child)
+
+  def parsepitch(self, xml_pitch):
+    """Parse the MusicXML <pitch> element"""
+    step = xml_pitch.find("step").text
+    alter = 0
+    if xml_pitch.find("alter") != None:
+      alter = xml_pitch.find("alter").text
+    octave = xml_pitch.find("octave").text
+
+    # Visual pitch representation
+    alter_string = ""
+    if alter == "-2":
+      alter_string = "bb"
+    elif alter == "-1":
+      alter_string = "b"
+    elif alter == "1":
+      alter_string = "#"
+    elif alter == "2":
+      alter_string = "x"
+
+    # N.B. - pitch_string does not account for transposition
+    pitch_string = step + alter_string + octave
+
+    # Compute MIDI pitch number (C4 = 60, C1 = 24, C0 = 12)
+    midi_pitch = self.pitchtomidipitch(step, alter, octave)
+    # Transpose MIDI pitch
+    midi_pitch = midi_pitch + self.state.transpose
+    self.pitch = (pitch_string, midi_pitch)
+
+  def parsetuplet(self, xml_time_modification):
+    """Parses a tuplet ratio, represented in MusicXML by the
+    <time-modification> element
+    """
+    numerator = int(xml_time_modification.find("actual-notes").text)
+    denominator = int(xml_time_modification.find("normal-notes").text)
+    self.note_duration.tuplet_ratio = Fraction(numerator, denominator)
+
+  @staticmethod
+  def pitchtomidipitch(step, alter, octave):
+    """Convert MusicXML pitch representation to MIDI pitch number"""
+    pitch_class = 0
+    if step == "C":
+      pitch_class = 0
+    elif step == "D":
+      pitch_class = 2
+    elif step == "E":
+      pitch_class = 4
+    elif step == "F":
+      pitch_class = 5
+    elif step == "G":
+      pitch_class = 7
+    elif step == "A":
+      pitch_class = 9
+    elif step == "B":
```

Comment:
There is no else branch, as all the possible valid cases are covered. 
http://usermanuals.musicxml.com/MusicXML/MusicXML.htm#EL-MusicXML-step.htm

The only thing an else branch could do would be to throw an exception. Would you like me to add that in this case? What kind of exception would it be?